classes/submission/Submission.inc.php

```diff
@@ -337,14 +374,20 @@ function getAuthorEmails() {
 	/**
 	 * Get all authors of this submission.
 	 * @param $onlyIncludeInBrowse boolean whether to limit to include_in_browse authors.
+	 * @param $version int
 	 * @return array Authors
 	 */
-	function getAuthors($onlyIncludeInBrowse = false) {
+	function getAuthors($onlyIncludeInBrowse = false, $version = null) {
 		$authorDao = DAORegistry::getDAO('AuthorDAO');
-		if (!$onlyIncludeInBrowse)
-			return $authorDao->getBySubmissionId($this->getId());
-		else
-			return $authorDao->getBySubmissionId($this->getId(), false, true);
+		if ($version == null) {
+			$version = $this->getCurrentVersionId() ? $this->getCurrentVersionId() : 1; // if a new submission is submitted, version is 1
+		}
+
+		if (!$onlyIncludeInBrowse) {
+			return $authorDao->getBySubmissionId($this->getId(), false, false, $version);
+		} else {
+			return $authorDao->getBySubmissionId($this->getId(), false, true, $version);
```

Comment:
should I change that although it is like this in the master branch? https://github.com/pkp/pkp-lib/blob/master/classes/submission/Submission.inc.php#L344-L347 
I agree it would be easier to read, but I am not sure how much I am allowed to change here (because you do not really need that for versioning...)